cmd/secret/cmd/root_test.go

```diff
@@ -123,3 +133,167 @@ func TestRoot_Get(t *testing.T) {
 	}
 	require.EqualValues(t, value, got)
 }
+
+func TestRoot_Server(t *testing.T) {
+	key := "key-value"
+	path := "testFile.txt"
+	t.Run("set by key", func(t *testing.T) {
+		t.Run("expect set method success", func(t *testing.T) {
+			expectedSipherKey := "key value"
+			ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+			defer cancel()
+			// get free port after cli creating
+			port := createAndExecuteCliCommand(ctx, t)
+
+			client := http.Client{Timeout: time.Second}
+			body := bytes.NewBufferString(`{"getter":"key-value","method":"local","value":"test-value-1"}`)
+			req := httptest.NewRequest(http.MethodPost, "http://localhost:"+port, body)
+			req.Header.Set(api.ParamCipherKey, expectedSipherKey)
+			req.RequestURI = ""
+
+			resp, err := client.Do(req)
+			require.NoError(t, err)
+			require.EqualValues(t, http.StatusNoContent, resp.StatusCode)
+			require.NoError(t, resp.Body.Close())
+
+			defer func() {
+				require.NoError(t, os.Remove(path))
+			}()
```

Comment:
This `defer` will not be executed if `client.Do()` returns error. Place it closer to where it is created.